src/app/pages/sharing/nvme-of/nvme-of-subsystems/nvme-of-subsystems.component.ts

```diff
@@ -1,12 +1,131 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  ChangeDetectionStrategy, Component, effect,
+} from '@angular/core';
+import { MatButton } from '@angular/material/button';
+import { MatDialog } from '@angular/material/dialog';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateModule } from '@ngx-translate/core';
+import {
+  filter, Observable, switchMap,
+  tap,
+} from 'rxjs';
+import { RequiresRolesDirective } from 'app/directives/requires-roles/requires-roles.directive';
+import { EmptyType } from 'app/enums/empty-type.enum';
+import { Role } from 'app/enums/role.enum';
+import { NvmeOfSubsystem } from 'app/interfaces/nvme-of.interface';
+import { ArrayDataProvider } from 'app/modules/ix-table/classes/array-data-provider/array-data-provider';
+import { SortDirection } from 'app/modules/ix-table/enums/sort-direction.enum';
+import { LoaderService } from 'app/modules/loader/loader.service';
+import { MasterDetailViewComponent } from 'app/modules/master-detail-view/master-detail-view.component';
+import { SlideIn } from 'app/modules/slide-ins/slide-in';
+import { TestDirective } from 'app/modules/test-id/test.directive';
+import { ApiService } from 'app/modules/websocket/api.service';
+import { AddSubsystemComponent } from 'app/pages/sharing/nvme-of/add-subsystem/add-subsystem.component';
+import { SubsystemDeleteDialogComponent } from 'app/pages/sharing/nvme-of/nvme-of-subsystems/subsystem-delete-dialog/subsystem-delete-dialog.component';
+import { SubsystemDetailsComponent } from 'app/pages/sharing/nvme-of/nvme-of-subsystems/subsystem-details/subsystem-details.component';
+import { SubsystemsListComponent } from 'app/pages/sharing/nvme-of/nvme-of-subsystems/subsystems-list/subsystems-list.component';
+import { NvmeOfStore } from 'app/pages/sharing/nvme-of/nvme-of.store';
 
+@UntilDestroy()
 @Component({
   selector: 'ix-nvme-of-subsystems',
   templateUrl: './nvme-of-subsystems.component.html',
   styleUrl: './nvme-of-subsystems.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
-  imports: [],
+  imports: [
+    TranslateModule,
+    SubsystemsListComponent,
+    MasterDetailViewComponent,
+    SubsystemDetailsComponent,
+    RequiresRolesDirective,
+    MatButton,
+    TestDirective,
+  ],
 })
 export class NvmeOfSubsystemsComponent {
+  protected dataProvider = new ArrayDataProvider<NvmeOfSubsystem>();
+  protected readonly subsystems = this.nvmeOfStore.subsystems;
+  protected readonly isLoading = this.nvmeOfStore.isLoading;
 
+  protected readonly requiredRoles = [
+    Role.SharingNvmeTargetWrite,
+    Role.SharingNvmeTargetWrite,
+    Role.SharingWrite,
+  ];
+
+  constructor(
+    private nvmeOfStore: NvmeOfStore,
+    private matDialog: MatDialog,
+    private slideIn: SlideIn,
+    private api: ApiService,
+    private loader: LoaderService,
+  ) {
+    effect(() => {
```

Comment:
I'm not a huge fan of effects, I think the make the flow of the program confusing. Perhaps we can make dataProvider a computed signal?